server/api.js

```diff
@@ -58,122 +60,151 @@ router.get("/callback", async (req, res) => {
-          res
-            .status(404)
-            .send("this is  github account does not belong to a CYF graduates");
-      }
-    );
-  });
-});
 
-router.get("/graduates/:id", (req, res, next) => {
+router.get("/graduates/:id", (req, res) => {
   const github_id = parseInt(req.params.id);
-  Connection.connect((err) => {
-    if (err) {
-      return next(err);
+  Connection.query(
+    "select g.*, s.skill_name from graduates g join graduate_skill gs on g.id=gs.graduate_id join skills s on s.id=gs.skill_id where g.github_id=$1",
+    [github_id],
+    (error, result) => {
+      if (result.rowCount > 0) res.json(result.rows);
+      else
+        res.status(404).send("It has not been added to the graduate table yet");
     }
-    Connection.query(
-      "SELECT * FROM graduates where github_id=$1 ",
-      [github_id],
-      (error, result) => {
-        if (result.rowCount > 0) res.json(result.rows);
-        else
-          res
-            .status(404)
-            .send("It has not been added to the graduate table yet");
-      }
-    );
-  });
+  );
 });
 
 //editing existing graduate
 router.put("/graduates/:id", function (req, res) {
-  const github_id = parseInt(req.params.id);
+  const github_id = req.params.id;
+  const avatar_url=req.body.avatar_url;
   const newFirstName = req.body.first_name;
   const newSurname = req.body.surname;
+  const aboutMe = req.body.about_me;
+  const location = req.body.location;
+  const interest = req.body.interest;
+  const github = req.body.github_link;
+  const linkedin = req.body.linkedin_link;
+  const portfolio = req.body.portfolio_link;
+  const skills =req.body.skills.map(x=>x.toLowerCase());
 
   Connection.query(
-    "update graduates set first_name=$1 ,surname=$2" + "where github_id =$3",
-    [newFirstName, newSurname, github_id],
-    (error) => {
-      if (error == undefined) {
-        res.send("graduate " + github_id + " updated.");
+    "update graduates set first_name=$1, surname=$2, about_me=$3, location=$4, interest=$5, github_link=$6, linkedin_link=$7, portfolio_link=$8 avatar_url=$9 where github_id =$10 returning id",
+    [
+      newFirstName,
+      newSurname,
+      aboutMe,
+      location,
+      interest,
+      github,
+      linkedin,
+      portfolio,
+      avatar_url,
+      github_id
+    ],
+    (error, result) => {
+      if (result) {
+        const graduate_id = result.rows[0].id
+        Connection.query(
+          "delete from graduate_skill where graduate_id=$1", [graduate_id],
+          (error,result)=>{
+            if(error) {
+              res.status(404).send(error)
+            } 
+          }
+        )
+        Connection.query(
+          `insert into graduate_skill (graduate_id, skill_id)` +
+           ` select $1, id from skills where skill_name=ANY($2)`, [graduate_id, skills],
+           (error, result)=>{
+              if (!error){
+                res.status(200).send('inserted succesfully')
+              }
+           }
```

Comment:
what happens if you have an error here? line 207 won't executed since it's a different branch, and you won't send anything back to the browser